pkg/kudo/operator.go

```diff
@@ -169,19 +174,31 @@ func (builder OperatorBuilder) Do(client client.Client) (Operator, error) {
 // We assume that this is the intended behavior for most test cases.
 // Don't use this for test cases which have multiple Instances for a single OperatorVersion.
 func (operator Operator) Uninstall() error {
+	return operator.UninstallWaitForDeletion(0)
+}
+
+// UninstallWaitForDeletion is the same as Uninstall but
+// initiates a foreground deletion, and waits for the KUDO resources to disappear.
+// Waits up to timeout for the instance to be deleted, and up to 10 seconds for each of OperatorVersion and Operator.
+func (operator Operator) UninstallWaitForDeletion(timeout time.Duration) error {
 	if operator.client.Kudo == nil {
 		return fmt.Errorf("operator is not initialized")
 	}
 
 	options := metav1.DeleteOptions{}
 
+	if timeout != 0 {
+		propagationPolicy := metav1.DeletePropagationForeground
```

Comment:
Foreground delete propagation is tricky. It uses finalizers and this caused issues with KUDO instances when the deletion is triggered before a deployment was completed (https://github.com/kudobuilder/kudo/issues/1071). Good that this isn't used if `timeout == 0` 👍. Let's document this caveat though, because this could result in hanging CI run if a deletion is triggered in failure conditions.

Reply:
Comment added.